public/app/features/alerting/unified/components/GrafanaAlertmanagerDeliveryWarning.test.tsx

```diff
@@ -1,39 +1,20 @@
 import { render, screen, waitFor } from '@testing-library/react';
-import { setupServer } from 'msw/node';
 import React from 'react';
 import { Provider } from 'react-redux';
 
-import { setBackendSrv } from '@grafana/runtime';
-import { backendSrv } from 'app/core/services/backend_srv';
+import { setupMswServer } from 'app/features/alerting/unified/mockApi';
+import { setAlertmanagerChoices } from 'app/features/alerting/unified/mocks/server/configure';
 import { configureStore } from 'app/store/configureStore';
 
 import { AlertmanagerChoice } from '../../../../plugins/datasource/alertmanager/types';
-import { mockAlertmanagerChoiceResponse } from '../mocks/alertmanagerApi';
 import { GRAFANA_RULES_SOURCE_NAME } from '../utils/datasource';
 
 import { GrafanaAlertmanagerDeliveryWarning } from './GrafanaAlertmanagerDeliveryWarning';
+setupMswServer();
 
 describe('GrafanaAlertmanagerDeliveryWarning', () => {
-  const server = setupServer();
-
-  beforeAll(() => {
-    setBackendSrv(backendSrv);
-    server.listen({ onUnhandledRequest: 'error' });
-  });
-
-  afterAll(() => {
-    server.close();
-  });
-
-  beforeEach(() => {
-    server.resetHandlers();
-  });
-
   it('Should not render when the datasource is not Grafana', () => {
-    mockAlertmanagerChoiceResponse(server, {
-      alertmanagersChoice: AlertmanagerChoice.External,
-      numExternalAlertmanagers: 0,
-    });
+    setAlertmanagerChoices(AlertmanagerChoice.External, 0);
```

Comment:
The important part being, when you look at it here:

* You aren't using a `server` 
* The inputs to this configuration method aren't a direct parallel to the corresponding API response. So you aren't conflating the configuration of the behaviour, with the API endpoints that it actually refers to under the hood